libbeat/outputs/redis/client.go

```diff
@@ -263,9 +271,17 @@ func (c *client) publishEventsPipeline(conn redis.Conn, command string) publishF
 			}
 
 			data = append(data, okEvents[i])
-			if err := conn.Send(command, eventKey, serializedEvent); err != nil {
-				logp.Err("Failed to execute %v: %v", command, err)
-				return okEvents, err
+			if command == "XADD" {
```

Comment:
What do you think about simplifying the code to this format?

```golang
var commandArgs []string
if command == "XADD" {
    commandArgs = []string{eventKey, "*", "event", serializedEvent}
} else {
    commandArgs = []string{eventKey, serializedEvent}
}
if err := conn.Send(command, commandArgs...); err != nil {
    logp.Err("Failed to execute %v: %v", command, err)
    return okEvents, err
}
```

This way `conn.Send` is not repeated.

Reply:
@kvch thanks that's much cleaner.  Updated